### PR TITLE
More stable covariance matrix calculation

### PR DESCRIPTION
**Pull request by janosg**
Replace inverses by pseudo inverses when necessary but produce a strong warning if standard inverse calculation would fail.

### CODE REVIEW

File: estimagic/optimization/utilities.py [hunk 1]
```diff
@@ -169,6 +172,34 @@ def robust_cholesky(matrix, threshold=None, return_info=False):
...
+    The difference to calling a pseudo inverse directly is that this function will
+    emit a warning if the matrix is non-singular.
+
+    Args:
+        matrix (np.ndarray)
+
+    """
+    header = (
+        "Standard matrix inversion failed due to LinAlgError described below. "
+        "Therefore a pseudo inverse was calculated instead. "
```
Comment (hmgaudecker) [inline estimagic/optimization/utilities.py:187]:
```suggestion
        "A pseudo inverse was calculated instead. "
```

**Review by hmgaudecker — APPROVED** [review]
Looks great! Just a couple of textual changes.

File: estimagic/inference/likelihood_covs.py [hunk 2]
```diff
@@ -1,6 +1,13 @@
 """Functions for inferences in maximum likelihood models."""
 import numpy as np
 
+from estimagic.optimization.utilities import robust_inverse
+
+LINALG_MSG = (
```
[the remark this answers is not in the document]
Comment (janosg) [reply estimagic/inference/likelihood_covs.py:6]: I will break up the `optimization.utilities` module completely in one of my next PRs so for now it is OK to leave it there.

File: estimagic/inference/moment_covs.py [hunk 3]
```diff
@@ -5,6 +5,9 @@
 """
 import numpy as np
 
+from estimagic.inference.likelihood_covs import LINALG_MSG
```
[the remark this answers is not in the document]
Comment (janosg) [reply estimagic/inference/moment_covs.py:8]: Done

**Comment by janosg** [comment]
Some people are still using it and it is a really low cost for us. In the current case the `convert_dtype` was just unnecessary.

**Pull request merged**